sui/deploy-gateway.js

```diff
@@ -107,6 +106,40 @@ async function processCommand(config, chain, options) {
     printInfo('Gateway deployed', JSON.stringify(contractConfig, null, 2));
 }
 
+async function upgradeGateway(chain, options, keypair, client) {
+    const contractsConfig = chain.contracts;
+    const packageName = 'axelar_gateway';
+    const builder = new TxBuilder(client);
+    const chainConfig = contractsConfig[packageName];
+    chainConfig.objects = chainConfig.objects || {};
+    chainConfig.objects.upgradeCap = options.upgradeCap || chainConfig.objects.UpgradeCap;
```

Comment:
this is confusing. why are there two cases for the same object key? why isn't the cap id stored during deployment to avoid requiring upgradeCap to be passed in?

Reply:
can you fix this deployment script to save the object?